api/v1alpha2/conversion.go

```diff
@@ -212,6 +215,7 @@ func (src *MachineDeployment) ConvertTo(dstRaw conversion.Hub) error {
 
 	if restored.Spec.ClusterName != "" {
 		dst.Spec.ClusterName = restored.Spec.ClusterName
+		dst.Spec.Template.Spec.ClusterName = restored.Spec.ClusterName
```

Comment:
This should not be necessary, should it? This is handled by the call to restoreMachineSpec.

Reply:
ah that's right, I forgot we added that other function. The restore from the label is still required though